public/app/features/alerting/unified/Silences.tsx

```diff
@@ -31,33 +38,65 @@ const Silences: FC = () => {
     };
   }, [alertManagerSourceName, dispatch]);
 
+  const { result, loading, error } = silences[alertManagerSourceName] || initialAsyncRequestState;
+  const getSilenceById = useCallback((id: string) => result && result.find((silence) => silence.id === id), [result]);
+
   if (!alertManagerSourceName) {
     return <Redirect to="/alerting/silences" />;
   }
-  const { result, loading, error } = silences[alertManagerSourceName] || initialAsyncRequestState;
 
   return (
     <AlertingPageWrapper pageId="silences">
       <Field label="Choose alert manager">
         <AlertManagerPicker current={alertManagerSourceName} onChange={setAlertManagerSourceName} />
```

Comment:
```suggestion
        <AlertManagerPicker current={alertManagerSourceName} onChange={setAlertManagerSourceName} disabled={!isRoot} />
```

Should disalbe the picker if user is not in root page. Otherwise it's weird if you change source while editing something... What would be the expectation there?

Reply:
I moved it to the Silences Table since that's really the only place where it should be used.